### DISCUSSION TITLE
programming style comments

### DISCUSSION THREAD

**Discussion by grondilu** [pystardust/ani-cli#116] [General] [answered]
Cool project.   I don't watch much animes these days but I love bash programming, so here are some programming style comments :

- variable `prog` could just be $0 or $BASH_SOURCE ;
- why the `while read` loop in help_text ?  Why a dedicated function anyway?
- in function `search_anime`, variable `search` could be defined as `search="${1// /-}"` ;
- in that same function, variable `title_pattern` is unused ;
- there is no declaration of variable whatsoever in the whole script (no `declare` or `local`) ;
- variables that are meant to be integers (e.g. episode numbers), could be defined as integer variables (`-i` option to `declare` or `local`);
- I think the standard way to create a file is with `touch`, not `: > "$filename"`;
- variables that are meant to contain several values (e.g. `seach_results`) might be defined as arrays;
- `select` is the standard way to get user input from a list of possible choices;
- not sure terminal colors are necessary, especially if they are the reasons `select` was not used;
- no `echo` at all, only `printf`?

**Comment by Dink4n** [comment, marked as answer]
First of all it's not a bash script it's a POSIX compliant shell script
That's why `select`, `declare`, `local` etc. are not used

* I don't know why `while read` is used in `help_text`. Isn't `printf` enough?
* Yes `title_pattern` is unused
* There are no arrays in POSIX compliant shell script

**Reply by grondilu** [reply]
Why make it POSIX?  It's a script for watching animes.  It really doesn't have to comply to industry standards.

I mean I used to want to conform to POSIX too so I think I get it, but then I realized that by doing so I deprived myself from all the nice  GNU features.  Your users are most likely GNU users, and GNU is Not Unix.

**Reply by port19x** [reply]
> Why make it POSIX? It's a script for watching animes. It really doesn't have to comply to industry standards.
> 
> I mean I used to want to conform to POSIX too so I think I get it, but then I realized that by doing so I deprived myself from all the nice GNU features. Your users are most likely GNU users, and GNU is Not Unix.

While I agree I come to the opposite conclusion: Its just a script for watching anime. Why would you need bashisms or GNU options?
POSIX compliance will also make the script easily portable preventing obscure issues from people using it on mac or windows, while also making life easier for anyone looking to port it to android or iOS in the future
